streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -972,8 +970,8 @@ public synchronized State state() {
         return state;
     }
 
-    private synchronized void setStateWhenNotInPendingShutdown(final State newState) {
-        if (state == State.PENDING_SHUTDOWN) {
+    private synchronized void setStateWhenNotShuttingDownOrDead(final State newState) {
+        if (state == State.PENDING_SHUTDOWN || state == State.DEAD) {
```

Comment:
One can call thread.close() many times, even when it's DEAD. So the check prevents any further changes when the thread is shutting down or dead.